go/boilerplate/materialize.go

```diff
@@ -0,0 +1,327 @@
+package boilerplate
```

Comment:
For reference, see https://github.com/estuary/flow/blob/master/go/protocols/materialize/transactor.go

Many of the comments and concepts in this file were lifted from there.